lib/bindings/python/rust/lib.rs

```diff
@@ -390,6 +524,33 @@ impl DistributedRuntime {
     }
 }
 
+// Bind a TCP port and return a socket held until dropped.
+fn bind_tcp_port(port: u16) -> std::io::Result<socket2::Socket> {
+    let sock = socket2::Socket::new(
+        socket2::Domain::IPV4,
+        socket2::Type::STREAM,
+        Some(socket2::Protocol::TCP),
+    )?;
+    sock.set_reuse_address(true)?;
+    let addr = SocketAddr::V4(SocketAddrV4::new(Ipv4Addr::UNSPECIFIED, port));
+    sock.bind(&addr.into())?;
+    Ok(sock)
```

Comment:
so, this function can only return success? 😕 

if so, then why return a result type at all?

Reply:
The `?` in Rust is shorthand for "if it's an error stop here and return it", so anywhere with that `?` is also a potential error return point.

Reply:
😔 I always miss that, thanks.